arviz/plots/dotplot.py

```diff
@@ -0,0 +1,142 @@
+import math
+import numpy as np
+
+from ..rcparams import rcParams
+from .plot_utils import get_plotting_function
+
+
+def plot_dots(
+    values=None,
+    binwidth=None,
+    dotsize=1,
+    stackratio=1,
+    hdi_prob=None,
+    rotated=False,
+    dotcolor="grey",
+    intervalcolor="red",
+    markersize=None,
+    figsize=None,
+    linewidth=None,
+    point_estimate="auto",
+    quantiles=None,
+    point_interval=None,
+    plot_kwargs=None,
+    interval_kwargs=None,
+    backend=None,
+    backend_kwargs=None,
+    **kwargs
+):
+
+    """Plot distribution as dot plot.
+
+    If quantiles is specified then it will plot the quantile dot plot.
+
+    Parameters
+    ----------
+    values : array-like
+        Values to plot
+    binwidth : float
+        Width of the bin for drawing the dot plot.
+    dotsize : float
+        The size of the dots relative to the bin width. The default, 1, makes dots be 
+        just about as wide as the bin width.
+    stackratio : float
+        The distance between the center of the dots in the same stack relative to the bin height. 
+        The default, 1, makes dots in the same stack just touch each other.
+    point_interval : bool
+        Plots the point interval. Uses hdi_prob to plot the HDI interval
+    point_estimate : Optional[str]
+        Plot point estimate per variable. Values should be ‘mean’, ‘median’, ‘mode’ or None. 
+        Defaults to ‘auto’ i.e. it falls back to default set in rcParams.
+    dotcolor : string
+        The color of the dots
+    intervalcolor : string
+        The color of the interval
+    linewidth : int
+        Line width throughout. If None it will be autoscaled based on figsize.
+    markersize : int
+        Markersize throughout. If None it will be autoscaled based on figsize.
+    hdi_prob : float
+        Valid only when point_interval is True. Plots HDI for chosen percentage of density. 
+        Defaults to 0.94.
+    rotated : bool
+        Whether to rotate the dot plot by 90 degrees.
+    quantiles : int
+        Number of quantiles to plot, used for quantile dot plots
```

Comment:
I think I am missing something. What is the alternative if quantiles is none?

Reply:
As I mentioned above it will set quantiles as the number of samples. Basically, it will use the actual data to plot. Otherwise, if quantiles are given and are less than the total number of samples then it would generate quantile number of points and use them to plot. This is done because if we randomly sample quantile number of points from our values and plot them then it may be that it won't represent the actual distribution. [This](https://github.com/mjskay/when-ish-is-my-bus/blob/master/quantile-dotplots.md) explains it much better.

Reply:
Should probably note that the minimum of this number and the number of data points is used for plotting.

Reply:
Yes, if quantile dot plot is set as default, so what should I set the default value of `quantiles` because 20 seems too little for me

Reply:
My understanding is that dot-plots are useful (or at leas more common) for small dataset, otherwise an histogram or KDE would be a better choice. And that quantile dot plots are generally used with a small number of quantiles because you expect people to actually be able to count the dots. So a small number like 20 seems like a reasonable default. But of course I may be missing something and there are other reasons to pick other default values.

Reply:
In http://users.eecs.northwestern.edu/~jhullman/busUncertaintyVis.pdf, which presented quantile dotplots, they found that users were more precise at estimating probabilities from quantile dotplots with 20 dots vs 100. On the other hand, in https://dl.acm.org/doi/10.1145/3173574.3173718 they found that when it came to decison making using quantile dotplots, users performed better with 50 dots vs 20 (they did not compare 100). Which would perhaps indicate that 50 is a reasonable default.